tests/test_app.py

```diff
@@ -0,0 +1,139 @@
+
+from threading import Thread
+
+#Allows one to use json values of null and false in a Python framework
+null = None
+false = False
+
+sample_trello_lists_response = [
+    {
+        "id": "5fdcda2428d71e74db64410e",
+        "checkItemStates": null,
+        "closed": false,
+        "dateLastActivity": "2020-12-18T16:34:44.809Z",
+        "desc": "",
+        "descData": null,
+        "dueReminder": null,
+        "idBoard": "555555555555555555555555",
+        "idList": "333333333333333333333333",
+        "idMembersVoted": [],
+        "idShort": 12,
+        "idAttachmentCover": null,
+        "idLabels": [],
+        "manualCoverAttachment": false,
+        "name": "Why does this not work",
+        "pos": 147456,
+        "shortLink": "pLobkTfQ",
+        "isTemplate": false,
+        "cardRole": null,
+        "badges": {
+            "attachmentsByType": {
+                "trello": {
+                    "board": 0,
+                    "card": 0
+                }
+            },
+            "location": false,
+            "votes": 0,
+            "viewingMemberVoted": false,
+            "subscribed": false,
+            "fogbugz": "",
+            "checkItems": 0,
+            "checkItemsChecked": 0,
+            "checkItemsEarliestDue": null,
+            "comments": 0,
+            "attachments": 0,
+            "description": false,
+            "due": null,
+            "dueComplete": false,
+            "start": null
+        },
+        "dueComplete": false,
+        "due": null,
+        "idChecklists": [],
+        "idMembers": [],
+        "labels": [],
+        "shortUrl": "https://trello.com/c/pLobkTfQ",
+        "start": null,
+        "subscribed": false,
+        "url": "https://trello.com/c/pLobkTfQ/12-why-does-this-not-work",
+        "cover": {
+            "idAttachment": null,
+            "color": null,
+            "idUploadedBackground": null,
+            "size": "normal",
+            "brightness": "light"
+        }
+    }
+]
+
+@pytest.fixture
+def client():
+    # Use out latest integration config instead of the 'real' version
+    file_path = find_dotenv('.env.test')
+    load_dotenv(file_path, override=True)
+
+    # Create the new app
+    test_app = app.create_app()
+
+    # Use the app to create a test_client that can be used in our tests
+    with test_app.test_client() as client:
+        yield client 
+
+@patch('requests.get')
+def test_index_page(mock_get_requests, client):
+    #Replace call to requests.get(url) with our own function
+    mock_get_requests.side_effects = mock_get_cards
+    response = client.get('/trello')
+    assert response.status_code == 200
+
+def mock_get_cards(url, params):
+    if url == f'https://api.trello.com/1/boards/{TRELLO_BOARD_ID}/lists':
+        response = Mock()
+        # sample_trello_lists_response should point to some test reponse data
+        response.json.return_value = sample_trello_cards_response
+        return response
+    return None
+'''
```

Comment:
Similarly from hereon down - this has all been moved to the end-to-end test, so we can get rid of it here (and similarly a couple of other places have commented out code that we can delete)